Makefile

```diff
@@ -296,8 +301,8 @@ test/test-rbuf: test/test-rbuf.o
 
 test/test-regidx.o: test/test-regidx.c $(htslib_kstring_h) $(htslib_hts_os_h) regidx.h
 
-test/test-regidx: test/test-regidx.o regidx.o $(HTSLIB)
-	$(CC) $(ALL_LDFLAGS) -o $@ $^ $(HTSLIB) -lpthread $(HTSLIB_LIB) $(ALL_LIBS)
+test/test-regidx: test/test-regidx.o regidx.o | $(HTSLIB)
```

Comment:
This shouldn't be order-only — _test/test-regidx_ still depends on it as usual. cf the _bcftools_ rule, which has `$(HTSLIB)` as a normal dependency and `$(HTSLIB_LIB)` in the recipe.

Reply:
I guess making this order-only was so that it wouldn't appear in `$^`, but the way to fix that while keeping the correct dependency is to write out `$^` explicitly as `test/test-regidx.o regidx.o` instead.